### PR TITLE
Updated to Swift 4.2

### PR DESCRIPTION
**Pull request by mattgwwalker**
Hi,

I see you haven't been involved with this project for a couple of years.  I've updated the project (minimal changes) to Swift 4.2.

Are you interested in further contributions?

Cheers,

Matthew

### CODE REVIEW

File: SwiftStats/stats.swift [hunk 1]
```diff
@@ -437,8 +437,8 @@ public struct Common {
 			let num = (((a[3]*z + a[2])*z + a[1])*z) + a[0]
 			let den = ((((b[3]*z + b[2])*z + b[1])*z + b[0])*z + 1.0)
 			var x = y*num/den
-			x = x - (erf(x) - y)/(2.0/sqrt(M_PI)*exp(-x*x))
-			x = x - (erf(x) - y)/(2.0/sqrt(M_PI)*exp(-x*x))
+			x = x - (erf(x) - y)/(2.0/sqrt(Double.pi)*exp(-x*x))
```
Comment (r0fls) [inline SwiftStats/stats.swift]: is there any reason we aren't doing this instead? Seems I defined the `pi` variable but don't use it here

```suggestion
			x = x - (erf(x) - y)/(2.0/sqrt(pi)*exp(-x*x))
```

**Comment by r0fls** [comment]
Hey @mattgwwalker thanks for the PR! I'm definitely seeking further contributions. If you are interested, I'd be happy to make you an official contributor (which will give you merge rights and the ability to manage issues).

**Comment by r0fls** [comment]
@mattgwwalker can you verify this was tested with 4.2? Did the unit tests pass?

One thing I really wanted to do with this project was get the travis tests passing -- there seemed to be some environment inconsistencies that wouldn't allow them to pass previously. Maybe we should open an issue about that so you and I can collaborate to make them work? That way it's more visible which changes are breaking for us as maintainers.

File: SwiftStats/stats.swift (hunk 1 above)
Comment (mattgwwalker) [reply SwiftStats/stats.swift]: I did see that you had defined a constant.  I went with the "make the fewest changes" option.  Some people really don't like it when others change their code--looks like you're happy to see contributions, which is great :o)

**Comment by mattgwwalker** [comment]
Here are a few more additions.  Yes, I'm testing under XCode 10.1, Swift 4.2, I'll add an automated test for that in my next pull request.

As for the travis tests, I can't see anything about them.  Would you like to raise that as an issue and we might be able to work on them together.

Also, I'd be very happy to become a contributor.

**Comment by mattgwwalker** [comment]
I have a number of other commits that I'd like to share with you, but I'm not quite sure how to go about that.  Perhaps you know more than I do?

I've created a new branch (add-lognormal) on my fork of your code, where I've added a LogNormal distribution and associated tests.  Should I:
(a) create a new pull request with everything in this pull request plus the log-normal code (but does that double up on the work you have to do?)
(b) wait till you accept the current pull request before creating another pull request
(c) keep on committing to my master branch on GitHub, in which case this pull request labelled "Updated to Swift 4.2" just gets larger and larger
(d) wait for you to set me up as a contributor, or
(e) do something else

Your assistance would be appreciated :o)

File: SwiftStats/stats.swift (hunk 1 above)
Comment (mattgwwalker) [reply SwiftStats/stats.swift]: I have fixed this in a recent commit.  I'm just waiting on you to tell me how you'd like pull requests.

**Comment by r0fls** [comment]
Hey @mattgwwalker thanks again for the contributions. Let's have you create the log normal pull request as a separate PR. So, option (b). I've merged this PR now so we should be good. I'm also going to add you as a contributor, but it's still ok to let me review your PRs if you like. I don't intend to let any of them sit around forever unless they're controversial or something, so if I happen to be overlooking anything just ping me.

**Comment by r0fls** [comment]
@mattgwwalker in general, it's best if you create separate pull requests for each change/addition or set of related changes. So, adding a new distribution would be one PR, refactoring code would be a separate one. Since this project is still pretty young, it's ok if we don't follow that strictly, but in general, that's the preferred method. 

To do that, essentially you'll want to create a new feature branch from master for each set of changes. Let me know if that clears things up or if you still have any questions.

**Pull request merged**